src/ncp/spinel.h

```diff
@@ -1175,10 +1174,43 @@ typedef char spinel_datatype_t;
 #define SPINEL_DATATYPE_IPv6ADDR_S    "6"
 #define SPINEL_DATATYPE_EUI64_S       "E"
 #define SPINEL_DATATYPE_EUI48_S       "e"
+#define SPINEL_DATATYPE_DATA_WLEN_S   "d"
 #define SPINEL_DATATYPE_DATA_S        "D"
 #define SPINEL_DATATYPE_UTF8_S        "U" //!< Zero-Terminated UTF8-Encoded String
-#define SPINEL_DATATYPE_STRUCT_S      "T"
-#define SPINEL_DATATYPE_ARRAY_S       "A"
+
+#define SPINEL_DATATYPE_ARRAY_S(x)        "A(" x ")"
+#define SPINEL_DATATYPE_STRUCT_S(x)       "t(" x ")"
+
+#define SPINEL_DATATYPE_ARRAY_STRUCT_S(x)                                     \
+        SPINEL_DATATYPE_ARRAY_S(SPINEL_DATATYPE_STRUCT_WLEN_S(x))
+
+#define SPINEL_DATATYPE_COMMAND_S                                             \
```

Comment:
I think the name of this define would be more accurately,
SPINEL_DATATYPE_HEADER_COMMAND_S 
and then down below instead of 
SPINEL_DATATYPE_COMMAND_PROP_S 
it would be
SPINEL_DATATYPE_HEADER_COMMAND_PROP_S.

Otherwise the presence of the header appears to be hidden by the name.